docs/styles/typography.css

```diff
@@ -4,50 +4,50 @@ body {
   line-height: var(--rh-line-height-body-text, 1.5);
 }
 
-:where(h1, h2, h3, h4, h5, h6) {
+:where(h1, h2, h3, h4, h5, h6):not([slot]) {
```

Comment:
I have questions about this approach.
 
Isn't it the components responsibility to ensure the correct h1-h6 styles are applied?  
Shouldn't the component apply a `!important` rather then saying the page styles don't apply to when the headers are in "lightdom"?
Is this a violation of those separation of concerns, as we won't be able to control other sites page styles but still want to ensure the components output the correct styling?

Reply:
Maybe if we add `:is([slot])` to the slotted selector we'll get a tiny specificity bump

Reply:
I'm still not sure why the specificity from the page css here would matter to the concerns laid out above.

If we are trying to enforce a specific style shouldn't this be the approach: https://codepen.io/zeroedin/pen/MWddJPQ

Reply:
#1701 might help here. I reverted these changes to page css, this pr will now focus on other improvements, like getting rid of the `data-demo` layout css. If individual demos want to put their elements in a padded container, that should be explicit in the demo css, rather than implicit in the dev server css